sql/data_prep.sql

```diff
@@ -141,4 +141,78 @@ SET direction1 =
     )
 
 -- results may be viewed at
--- https://jbranigan.carto.com/builder/0739a750-359f-11e7-b40d-0ee66e2c9693/embed
\ No newline at end of file
+-- https://jbranigan.carto.com/builder/0739a750-359f-11e7-b40d-0ee66e2c9693/embed
+
+--==================================================================
+--==================================================================
+--==================================================================
+
+-- On to trying to fix the topology
+-- The topology extension is not available in CARTO
+-- So going to try to snap endpoints in a copy of ecglines_clean, called ecglines_topo
+
+-- This requires LineString types, so checking multilines
+SELECT COUNT(CASE WHEN ST_NumGeometries(the_geom_webmercator) > 1 THEN 1 END) AS multi_geom,
```

Comment:
@clhenrick - Regarding your multigeometry comment, this query checks to see if there are actually multilines in the dataset. The result looks like every line is simple geometry, so the conversion is safe.